lib/src/edhoc.rs

```diff
@@ -744,11 +759,22 @@ fn decrypt_message_3(
     message_3: &BufferMessage3,
 ) -> Result<BufferPlaintext3, EDHOCError> {
     // decode message_3
-    let len = (message_3.content[0usize] ^ CBOR_MAJOR_BYTE_STRING) as usize;
+    let bytestring_length: usize;
+    let prefix_length;
+    // FIXME: Reuse CBOR decoder
+    if (0..=23).contains(&(message_3.content[0] ^ CBOR_MAJOR_BYTE_STRING)) {
+        bytestring_length = (message_3.content[0] ^ CBOR_MAJOR_BYTE_STRING).into();
+        prefix_length = 1;
+    } else {
+        // FIXME: Assumes we don't exceed 256 bytes which is the current buffer size
+        bytestring_length = message_3.content[1].into();
+        prefix_length = 2;
+    }
```

Comment:
Same as above.

Reply:
See https://github.com/openwsn-berkeley/lakers/pull/267#discussion_r1601247211 -- can change it if you insist, but it was a conscious choice. Here, the alternative is `let (bytestring_length, prefix_length) = match { ... (message_3.content[1].into(), 2) ...}` that is even less clear to the reader -- that was actually what prompted me to use the assign-once variables in both cases.

Reply:
Sounds good.